simpleviewinc/tap-events-force
# Alert modal

> NOTE: depends on PR https://github.com/simpleviewinc/keg-hub/pull/38 and new keg-core release
[Semver Status] major
     - instead of having to pass each property individually. we can now pass in the agenda props to `sessionAgendaProps`
     - this name matches the interface given by eventsForce
     - will be more scalable for when they want to add more property to the `EFSessionAgendaProps` interface

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-292) 

## Context

* Discussed with Eventsforce, they will pass in an `alert` prop to the sessions component
* this is so they can dynamically display an alert on render
* we need to check this prop if it exists  && if it has valid content then add a new alert modal to display

## Goal

* utilize the `alert` prop passed in so we can decide whether to display some kind of alert on re-render or not
* update `Sessions` props to be more explicit

## Updates

* `src/actions/session/mapSessionInterface.js`
   * updated mapSessioninterface to check for passed in alert and call `addModal` if it does
   * no need to store `alert` in redux store
* `src/components/modals/alert.js`, `src/components/modals/renderModals.js`
  * renamed the modal `error` to `alert`

* `src/containers/sessionsContainer.js`
   * updated the session props to be more explicit, using the same name as the given EventsForce interface

* `src/containers/rootContainer.js`, `src/containers/testContainer.js`
   * added `apply` button for JSON test dat
   * updated the JSON data input to only re render when Apply is pressed. it was re rendering everytime you type a character. which was annoying
## Testing
> Note: this package build is synced with https://github.com/simpleviewinc/keg-hub/pull/38
* Run `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:alert-modal`
* navigate to http://local.kegdev.xyz/
* verify that you see a modal pop up on render
* close the modal
* search for `"alert"` in the JSON inputbox
* update the `title` and `message` property
* hit `Apply`
* verify that a modal appears with the given content

* **build**: run `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:alert-modal command=test:build`
